src/components/CheckboxLegacy/CheckboxLegacy.js

```diff
@@ -39,14 +41,21 @@ const CheckboxLegacy = ({
           e.stopPropagation();
           onClickFn(e, disabled, setIsChecked);
         }}
-        role='checkbox'
         tabIndex='-1'
         aria-checked={isChecked}
         checked={isChecked}
         aria-label={children}
         className={className}
         $disabled={disabled}
+        data-testid='checkbox-legacy'
```

Comment:
all kinds of IDs should be unique per page. with this here we might very easily have lots of the same IDs on one page. it would be good to make this more specific. it could have the `id` value

also, since we pass `id` in props, do we even need this attribute?

Reply:
Yup but currently in other places than `Adyen`  this `CheckboxLegacy` doesn't have any `id` passed as a prop, so for other tests to pass I would have to add those ids and more precisely change tests according to them. I can do it in this task but it will require more time.